private/bufpkg/bufmodule/module.go

```diff
@@ -364,6 +365,13 @@ func (m *module) Commit() string {
 	return m.commit
 }
 
+func (m *module) ID() string {
+	if m.moduleIdentity != nil {
+		return "mod:" + m.moduleIdentity.IdentityString()
+	}
+	return m.id
+}
```

Comment:
If we change the `ID` to come from the ref, then we won't need this default, since any module built with that option will have the ref type + `subDir` as the ID.